src/rgw/driver/rados/rgw_trim_bilog.cc

```diff
@@ -769,6 +855,15 @@ int BucketTrimInstanceCR::operate(const DoutPrefixProvider *dpp)
 	return set_cr_error(-EINVAL);
       }
 
+      if (clean_info->first.layout.logs.front().layout.type == rgw::BucketLogType::Deleted) {
+        yield call(new ListBucketCR(cct, store, bucket_instance, &list_results));
+        if (retcode < 0) {
+          ldpp_dout(dpp, 10) << "failed to list bucket: " << bucket_instance
+            << cpp_strerror(retcode) << dendl;
+          return set_cr_error(retcode);
```

Comment:
i don't think these functions have to return error. they are not strictly necessary for trim to be successful.